huaweicloud/services/cfw/resource_huaweicloud_cfw_eip_protection.go

```diff
@@ -0,0 +1,423 @@
+		PollInterval: 10 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("the EIPs to be protected have not been fully synchronized to CFW: %s", err)
+	}
+
+	opts := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes:          []int{200},
+	}
+
+	path := client.Endpoint + modifyProtectHttpUrl
+	path = strings.ReplaceAll(path, "{project_id}", client.ProjectID)
+	opts.JSONBody = utils.RemoveNil(requestParams)
+	_, err = client.Request("POST", path, &opts)
+	if err != nil {
+		return fmt.Errorf("error opening EIP protection: %s", err)
+	}
+
+	// After modifying the EIP protection, check the status of all protected public IPs.
+	stateConf = &resource.StateChangeConf{
+		Pending:      []string{"PENDING"},
+		Target:       []string{"COMPLETED"},
+		Refresh:      syncedEipsRefreshFunc(client, queryHttpUrl, objectId, protectedEipIds, openEipProtection),
+		Timeout:      timeout,
+		Delay:        5 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("an error occurred while protecting EIPs: %s", err)
+	}
+	return nil
+}
+
+func resourceEipProtectionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg    = meta.(*config.Config)
+		region = cfg.GetRegion(d)
+
+		objectId      = d.Get("object_id").(string)
+		protectedEips = d.Get("protected_eip").(*schema.Set)
+	)
+	client, err := cfg.NewServiceClient(eipProtectionProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating CFW Client: %s", err)
+	}
+
+	err = protectEips(ctx, client, objectId, protectedEips, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(objectId)
+
+	return resourceEipProtectionRead(ctx, d, meta)
+}
+
+func buildModifyProtectedEipsParams(objectId string, protectedEips *schema.Set,
+	operation int) (map[string]interface{}, []string) {
+	protectEipInfo, protectedEipIds := buildProtectedEipBodyAndIds(protectedEips)
+	requestParams := map[string]interface{}{
+		"object_id": objectId,
+		"ip_infos":  protectEipInfo,
+		"status":    operation,
+	}
+	return requestParams, protectedEipIds
+}
+
+func buildProtectedEipBodyAndIds(rawParams *schema.Set) ([]map[string]interface{}, []string) {
+	if rawParams.Len() < 1 {
+		return nil, nil
+	}
+
+	requestParams := make([]map[string]interface{}, rawParams.Len())
+	protectedEipIds := make([]string, rawParams.Len())
+	for i, v := range rawParams.List() {
+		raw := v.(map[string]interface{})
+		requestParams[i] = map[string]interface{}{
+			"id":          raw["id"],
+			"public_ip":   utils.ValueIngoreEmpty(raw["public_ipv4"]),
+			"public_ipv6": utils.ValueIngoreEmpty(raw["public_ipv6"]),
+		}
+		protectedEipIds[i] = raw["id"].(string)
+	}
+	return requestParams, protectedEipIds
+}
+
+// NoProtectedEips method check whether exist unprotected public IPs.
+func NoProtectedEips(eips []interface{}) bool {
```

Comment:
How about rename this function name to ProtectedEipExist

Reply:
NoProtectedEipsExist() maybe better.

Reply:
seems be ok

Reply:
done.